internal/gatewayapi/clienttrafficpolicy.go

```diff
@@ -359,3 +362,45 @@ func translateListenerSuppressEnvoyHeaders(suppressEnvoyHeaders *bool, httpIR *i
 		httpIR.SuppressEnvoyHeaders = *suppressEnvoyHeaders
 	}
 }
+
+func translateListenerTLSParameters(tlsParams *egv1a1.TLSSettings, httpIR *ir.HTTPListener) {
+	// Return if this listener isn't a TLS listener. There has to be
+	// at least one certificate defined, which would cause httpIR to
+	// have a TLS structure.
+	if httpIR.TLS == nil {
+		return
+	}
+	// Make sure that the negotiated TLS protocol version is as expected if TLS is used,
+	// regardless of if TLS parameters were used in the ClientTrafficPolicy or not
+	httpIR.TLS.MinVersion = ptr.To(ir.TLSv12)
+	httpIR.TLS.MaxVersion = ptr.To(ir.TLSv13)
+	// If HTTP3 is enabled, the ALPN protocols array should be hardcoded
+	// for HTTP3
+	if httpIR.HTTP3 != nil {
+		httpIR.TLS.ALPNProtocols = []string{"h3"}
+	} else if tlsParams != nil {
+		httpIR.TLS.ALPNProtocols = make([]string, len(tlsParams.ALPNProtocols))
```

Comment:
shouldnt this line 
```
httpIR.TLS.ALPNProtocols = make([]string, len(tlsParams.ALPNProtocols))
```
be under `if lsParams.ALPNProtocols != nil`

else it will end up resetting the value

Reply:
Fixed.